### PR TITLE
removed support for dynamic user generated visit labels

### PR DESCRIPTION
**Pull request by ridz1208**
This removes the possibility to enter a custom visit label dynamically when creating a new timepoint. This is to lay the foundation of normalised visits and help the move of visitLabels from the config.xml to the database.

collaterals: the suggestion options that was custom to user generated visitlabels was removed, Deprecation error is thrown id config.xml not updated before going on this branch

reqs: removing the `generation`, `regex`, `length` and `suggest` tags from the config.xml under the `visitLabel` section.

To test:
- first, before checking out this branch, set the generation to user and confirm that you can enter the visit label manually.
- checkout the branch and make sure an error is thrown when config is not modified
- change the generation=user to generation = sequence, this should not throw an error but we still recommend to completely remove the tags (this is just to simplify the upgrade for projects already using sequence)
- completely remove the tags mentioned above and go through the testplan.md file

### CODE REVIEW

**Comment by ridz1208** [comment]
@driusan why does this look like its failing travis but there are no errors in the log

**Comment by driusan** [comment]
There's a test failing with 7.0, and it looks like there was a network error so it wouldn't download selenium for 7.1. I don't know what you mean there's no errors.

**Comment by xlecours** [comment]
@ridz conflicts

**Comment by ridz1208** [comment]
@xlecours @driusan fixed

**Comment by ridz1208** [comment]
@PapillonMcGill I rebased this one if you want to take a look at it

File: modules/create_timepoint/php/create_timepoint.class.inc [hunk 1]
```diff
@@ -149,34 +136,24 @@ class Create_Timepoint extends \NDB_Form
         $visitLabelAdded    = false;
 
         foreach (\Utility::toArray($visitLabelSettings) as $visitLabel) {
-            if ($visitLabel['@']['subprojectID']==$this->subprojectID) {
-                if ($visitLabel['generation'] == 'user') {
-                    error_log(
-                        "DEPRECATION WARNING: User generated visit labels will ".
```
Comment (driusan) [inline modules/create_timepoint/php/create_timepoint.class.inc]: what version was this deprecation warning added in? Have people had sufficient time to react?

Comment (ridz1208) [reply modules/create_timepoint/php/create_timepoint.class.inc]: i think its been a couple years

Comment (ridz1208) [reply modules/create_timepoint/php/create_timepoint.class.inc]: https://github.com/aces/Loris/pull/2451

**Comment by ridz1208** [comment]
@driusan @PapillonMcGill 
visits normalization can not go through without this being merged first

**Comment by ridz1208** [comment]
@driusan help, not sure which phan rule is breaking

**Comment by driusan** [comment]
@ridz1208 Rebase, it's something that was broken in the major branch and is fixed now, it's not from this code..

```
modules/dicom_archive/php/dicomarchiveanonymizer.class.inc:42 PhanTypeMismatchDeclaredParam Doc-block of $resource in map contains phpdoc param type \LORIS\Data\Instance which is incompatible with the param type \LORIS\Data\DataInstance declared in the signature
```

(there was a typo where the doc class type wasn't renamed when the class was renamed from Instance to DataInstance..)

**Pull request merged**